raiden/tests/fixtures/api.py

```diff
@@ -15,11 +15,15 @@
 from raiden.network.discovery import Discovery
 from raiden.tests.utils.apitestcontext import ApiTestContext
 
+_SELF = object()
 
-def wait_for_listening_port(port_number, tries=10, sleep=0.1):
+
+def wait_for_listening_port(port_number, tries=10, sleep=0.1, pid=_SELF):
+    if pid is _SELF:
+        pid = os.getpid()
     for _ in range(tries):
         gevent.sleep(sleep)
-        connections = psutil.net_connections()
+        connections = psutil.Process(pid).connections()
```

Comment:
This [filters the connections](http://psutil.readthedocs.io/en/latest/#psutil.Process.connections) for the given `pid`, why?

I'm guessing macos requires root access for the connections api to work, is that it?

Reply:
@ulope Is Augusto right?